src/main/java/net/ccbluex/liquidbounce/ui/client/hud/element/elements/Text.kt

```diff
@@ -62,19 +64,19 @@ class Text(x: Double = 10.0, y: Double = 10.0, scale: Float = 1F, side: Side = S
 
     private var displayString by TextValue("DisplayText", "")
 
-    private val rainbow by BoolValue("Rainbow", false)
-    private val rainbowX by FloatValue("Rainbow-X", -1000F, -2000F..2000F) { rainbow }
-    private val rainbowY by FloatValue("Rainbow-Y", -1000F, -2000F..2000F) { rainbow }
+    private val textColor by ListValue("TextColor", arrayOf("Custom", "Rainbow"), "Custom")
+    private val rainbowX by FloatValue("Rainbow-X", -1000F, -2000F..2000F) { textColor == "Rainbow" }
+    private val rainbowY by FloatValue("Rainbow-Y", -1000F, -2000F..2000F) { textColor == "Rainbow" }
 
-    private var red by IntegerValue("Red", 255, 0..255) { !rainbow }
-    private var green by IntegerValue("Green", 255, 0..255) { !rainbow }
-    private var blue by IntegerValue("Blue", 255, 0..255) { !rainbow }
-    private var alpha by IntegerValue("Alpha", 255, 0..255) { !rainbow }
+    private var alpha by IntegerValue("Alpha", 255, 0..255) { textColor == "Custom" }
+    private var red by IntegerValue("Red", 255, 0..255) { alpha > 0 }
```

Comment:
rgb values don't check if textColor == "Custom"

should preferably be
alphaValue.isSupported() && !alphaValue.isMinimal()

Reply:
thanks!